src/node/factory.ts

```diff
@@ -14,6 +14,133 @@
  * You should have received a copy of the GNU Affero General Public License
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
+import {
+  ArweaveSigner,
+  EthereumSigner,
+  HexSolanaSigner,
+  JWKInterface,
+} from 'arbundles';
+
 import { TurboBaseFactory } from '../common/factory.js';
+import { defaultTokenMap } from '../common/index.js';
+import { TurboAuthenticatedPaymentService } from '../common/payment.js';
+import { TurboDataItemAbstractSigner } from '../common/signer.js';
+import { TurboAuthenticatedClient } from '../common/turbo.js';
+import {
+  TokenType,
+  TurboAuthenticatedConfiguration,
+  TurboSigner,
+  TurboWallet,
+  isEthPrivateKey,
+  isJWK,
+} from '../types.js';
+import { isWeb } from '../utils/common.js';
+import { TurboWebArweaveSigner } from '../web/signer.js';
+import { TurboNodeSigner } from './signer.js';
+import { TurboAuthenticatedUploadService } from './upload.js';
+
+export class TurboFactory extends TurboBaseFactory {
+  protected static getSigner(
+    providedSigner: TurboSigner | undefined,
+    providedPrivateKey: TurboWallet | undefined,
+    token: TokenType,
+  ): TurboDataItemAbstractSigner {
+    let signer: TurboSigner;
+
+    if (providedSigner !== undefined) {
+      signer = providedSigner;
+    } else if (providedPrivateKey !== undefined) {
+      if (token === 'solana') {
+        signer = new HexSolanaSigner(providedPrivateKey);
+      } else if (token === 'ethereum') {
+        if (!isEthPrivateKey(providedPrivateKey)) {
+          throw new Error(
+            'An Ethereum private key must be provided for EthereumSigner.',
+          );
+        }
+        signer = new EthereumSigner(providedPrivateKey);
+      } else {
+        if (!isJWK(providedPrivateKey)) {
+          throw new Error('A JWK must be provided for ArweaveSigner.');
+        }
+        signer = new ArweaveSigner(providedPrivateKey as JWKInterface);
+      }
+    } else {
+      throw new Error('A privateKey or signer must be provided.');
+    }
+
+    if (isWeb()) {
+      return new TurboWebArweaveSigner({
+        signer,
+        logger: this.logger,
+      });
+    }
```

Comment:
same here - since we're in the node export, all these should only concern themselves with the node configuration/types